service/ProductManager.java

```diff
@@ -0,0 +1,66 @@
+package service;
+
+import exceptions.DuplicateProductException;
+import exceptions.ProductNotFoundException;
+import model.Product;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProductManager {
+    private List<Product> inventory;
+
+    public ProductManager() {
+        inventory = new ArrayList<>();
+    }
+
+    public void addProduct(Product product) throws DuplicateProductException {
+        for (Product p : inventory) {
+            if (p.getId() == product.getId()) {
+                throw new DuplicateProductException("Product with ID " + product.getId() + " already exists.");
+            }
+        }
+        inventory.add(product);
+    }
+
+    public void updateProduct(Product product) throws ProductNotFoundException {
+        boolean found = false;
```

Comment:
maybe you can make a function to find the product as we are checking for duplicate products in multiple functions. so it'll remove code repetition.

Reply:
Done